### PR TITLE
SW-5322 Removed events from module store, added events store and new controllers

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2335?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2335** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2335?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2334** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2334?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#2337](https://github.com/terraware/terraware-server/pull/2337) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2337?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2338](https://github.com/terraware/terraware-server/pull/2338) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2338?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2332** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2332?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2333](https://github.com/terraware/terraware-server/pull/2333) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2333?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#2331** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2331?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2330** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2330?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/accelerator/api/EventsController.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.net.URI
+import java.time.Instant
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/events/")
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/EventsController.kt:24]: The trailing slash here might have unwanted effects, e.g., `GET /api/v1/events` won't work.

File: src/main/kotlin/com/terraformation/backend/accelerator/api/EventsController.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+@RestController
+class EventsController(
+    private val eventStore: ModuleEventStore,
+) {
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping
+  @Operation(summary = "list events")
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/EventsController.kt:32]: Nit: Capitalize this for consistency with other summaries (and maybe expand on it slightly?)

File: src/main/kotlin/com/terraformation/backend/accelerator/api/ModulesController.kt [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import java.time.InstantSource
+import java.time.LocalDate
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@AcceleratorEndpoint
+@RequestMapping("/api/v1/modules/")
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/ModulesController.kt:26]: Same comment as above about trailing slashes.

File: src/main/kotlin/com/terraformation/backend/accelerator/api/ModulesController.kt [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                participantId = participantId,
+                cohortId = cohortId,
+                moduleId = moduleId)
+            .firstOrNull() ?: throw ModuleNotFoundException(moduleId)
+    val today = LocalDate.ofInstant(clock.instant(), TimeZones.UTC)
+    return GetModuleResponsePayload(Module(model, today))
+  }
+}
+
+data class Module(
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/ModulesController.kt:71]: Maybe rename this to `ModulePayload` so it doesn't get auto-imported by accident if someone is using autocomplete to find the module model class and isn't paying close attention to what gets matched.

**Review by sgrimm — CHANGES_REQUESTED** [review]
This is a good change in general; I just have a few small comments.

Splitting things up like this at the API level is much cleaner. The original API design was mostly born out of a desire to minimize the amount of stitching together of data from multiple requests that had to happen on the client side, but it definitely made things more complex on the server side.

**Comment by tommylau523** [comment]
I am closing this branch because this is an unwanted split. I will address the comment in #2338

**Pull request closed without merging**